### PR TITLE
learnr support

### LINKED ISSUE #9: Create a sortable question type for `learnr`
The dev version of `learnr` has support for custom question types.

Create a `sortable item list` question type for `learnr`

Copying @schloerke

### PR DESCRIPTION
**Pull request by schloerke**
Fixes #9
Fixes rstudio/learnr#236

* Used snake case on all the functions 
  * The pkg is not on cran, felt like this was an OK change.
* added `options` to `sortable_list`
* added `learnr` compliant methods (forces an imports dependency)
* added manual `lintr` support
* set up travis file
  * Needs to be enabled
* passed `devtools::check()`
* added basic `learnr` `sortable` tutorial. Needs more content.

The jury is still out on how `learnr::question` will accept extra options.  For now, using `modifyList` to add `options`.

cc @andrie

### CODE REVIEW

**Comment by timelyportfolio** [comment]
@schloerke I'll transfer rights.  Let me know how I can help.

**Comment by schloerke** [comment]
Mind making me an admin? Should be able to solve it from there.

Thank you so much!!

**Comment by timelyportfolio** [comment]
@schloerke just set you up as a collaborator so you should see an invite.  Let me know if I need to do anything else.

**Comment by schloerke** [comment]
Will do!  Thank you again!

**Comment by andrie** [comment]
When I try to `devtools::load_all()` I get this error and warning:

```
Error: object 'question_completed_input' not found whilst loading namespace 'sortable'
In addition: Warning messages:
1: object ‘question_completed_input’ is not exported by 'namespace:learnr' 
2: object ‘question_initialize_input’ is not exported by 'namespace:learnr' 
3: object ‘question_is_correct’ is not exported by 'namespace:learnr' 
4: object ‘question_is_correct_value’ is not exported by 'namespace:learnr' 
5: object ‘question_is_valid’ is not exported by 'namespace:learnr
```

**Comment by schloerke** [comment]
Make sure to run devtools::install() before.

**Comment by andrie** [comment]
Sorry, you can ignore this report.  I did install using `devtools`, but I later discovered a corrupt installation, so using `devtools` with `force = TRUE` fixed this.

File: R/sortable_options.R [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+#' @param onSort JS function called by any change to the list (add / update / remove)
+#' @param onRemove JS function called when an element is removed from the list into another list
+#' @param onFilter JS function called when an attempt is made to drag a filtered element
+#' @param onMove JS function called when an item is moved in a list or between lists
+# ' @param onClone JS function that is called when creating a clone of an element
+# ' @param onChange JS function that is called when a dragging element changes position
+#' @param ... other params passed onto Sortable.js
+#' @seealso [https://github.com/sortablejs/Sortable/](https://github.com/sortablejs/Sortable/)
+#' @export
+sortable_options <- function(
```
Comment (andrie) [inline R/sortable_options.R]: This function is an excellent idea, thank you!

File: R/sortable_options.R [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  # merge all args
+  ret <- append(arg_vals, extra_args)
+
+  class(ret) <- "sortable_options"
+  ret
+}
+
+
+
+assert_sortable_options <- function(x) {
```
Comment (andrie) [inline R/sortable_options.R]: Let's use the `assertthat` package for assertions.

File: R/learnr.R [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+#' if (require(learnr, quietly = TRUE)) {
+#'   # to be used within a learnr tutorial...
+#'   question_sortable(
+#'     "Sort the first 5 letters",
+#'     answer(letters[1:5], correct = TRUE),
+#'     allow_retry = TRUE,
+#'     options = sortable_options(animation = 150)
+#'   )
+#' }
+question_sortable <- function(..., options = sortable_options()) {
```
Comment (andrie) [inline R/learnr.R]: I think we should explicitly set `random_answer_order = TRUE` and pass this on to `learnr`.

File: R/sortable_options.R (hunk 2 above)
Comment (schloerke) [reply R/sortable_options.R]: For now, we should pass as it's only checking for a class.  If we do actual type checking, my go to is `checkmate` and I believe it should be done inside `sortable_options`.

File: R/learnr.R (hunk 3 above)
Comment (schloerke) [reply R/learnr.R]: Added

**Pull request merged**